log/api.go

```diff
@@ -45,6 +44,21 @@ func Infof(ctx context.Context, format string, args ...interface{}) {
 	Fields{}.Infof(ctx, format, args...)
 }
 
+// RegisterListener registers callbacks that are called after a log.
+func RegisterListener(ctx context.Context, callback func(context.Context, Fields)) context.Context{
+	cbList, isList := ctx.Value(listenerKey{}).([]func(context.Context, Fields))
```

Comment:
Is `listenerKey{}` guaranteed to be there?

Reply:
No, but when it doesn't, it will define a new list and add it to the context with `listenerKey{}`. Added a test for this.

Reply:
nvm, I was reading it wrong.